tools/clang/lib/SPIRV/InitListHandler.cpp

```diff
@@ -118,25 +121,26 @@ bool InitListHandler::tryToSplitStruct() {
   if (initializers.empty())
     return false;
 
-  auto *init = initializers.back();
+  auto *init = initializers.back().first;
   const QualType initType = init->getAstResultType();
   if (!initType->isStructureType() ||
       // Sampler types will pass the above check but we cannot split it.
       isSampler(initType))
     return false;
 
   // We are certain the current intializer will be replaced by now.
+  const auto loc = initializers.back().second;
   initializers.pop_back();
 
   const auto *structDecl = initType->getAsStructureType()->getDecl();
 
   // Create MemberExpr for each field of the struct
-  llvm::SmallVector<SpirvInstruction *, 4> fields;
+  llvm::SmallVector<std::pair<SpirvInstruction *, SourceLocation>, 4> fields;
   uint32_t i = 0;
   for (auto *field : structDecl->fields()) {
     auto *extract =
-        spvBuilder.createCompositeExtract(field->getType(), init, {i});
-    fields.push_back(extract);
+        spvBuilder.createCompositeExtract(field->getType(), init, {i}, loc);
```

Comment:
Sorry. You're right, we cannot use `field->getSourceRange().getBegin()` because that will give us the struct field location.
What I really intended to say was: can we use `init->getSourceLocation()` ?

Reply:
done